tools/torch_quant/torch_quant/quantizer.py

```diff
@@ -0,0 +1,57 @@
+from enum import Enum
+from typing import Optional
+
+import torch.nn as nn
+from torch.fx import GraphModule, Tracer
+
+from torch_quant.graph import (fold_qdq, insert_act_observer, modify_graph,
+                               observer_to_qdq, q_ref_dq_to_fbgemm,
+                               quantizable_module_to_ref, set_qconfig)
+from torch_quant.module import ModuleFilter, copy_and_replace, fx_trace
+from torch_quant.observer import toggle_observer
+
+
+class Backend(Enum):
+    REFERENCE = 0
+    DISC = 1
+    FBGEMM = 2
+
+
+class Quantizer:
+    def __init__(self, module_filter: Optional[ModuleFilter] = None,
+                 backend: Optional[Backend] = Backend.REFERENCE,
+                 tracer: Optional[Tracer] = None) -> None:
+        self.module_filter = module_filter
+        self.backend = backend
+        self.tracer = tracer
+
+    def calib_gm(self,  gm: GraphModule, root: nn.Module) -> None:
+        modify_graph(gm, root, [set_qconfig, insert_act_observer])
+        toggle_observer(gm, observe=True, fake_quant=False)
+
+    def calib(self, model: nn.Module) -> nn.Module:
+        trace_mapping = fx_trace(model, self.module_filter, tracer=self.tracer)
+        for x in trace_mapping.values():
```

Comment:
why for-loop the `trace_mapping.values()` here

Reply:
The `fx_trace()` will support specifying module filters: user can explicitly include or exclude some modules in root model.  so the returned value of `fx_trace()` could be a dict like:

```
{
    'sub_module.conv_layer_0': TracePair(gm=..., m=...),
    'sub_module.conv_layer_1': TracePair(gm=..., m=...),
}
```

As for now (without filters), returned value is just `{'': TracePair(gm=..., m=root)}`